docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,857 @@
+    - [validUrl(url)](#validUrl)
+    - [delay(ms)](#delay)
+  - [middleware.js](#middlewarejs)
+    - [Environment Variables](#envvarsmiddleware)
+    - [readState()](#readState)
+    - [initialiseState()](#initialiseState)
+    - [getJobIndex(jobName)](#getJobIndex)
+    - [sendJobRun(credentials, count, jobId, chainlinkUrl, requestType)](#sendJobRun)
+    - [getOffers(follower)](#getOffers)
+    - [checkSubmissionForRound(oracle, feedOfferId, roundId)](#checkSubmissionForRound)
+    - [queryPrice(feed)](#queryPrice)
+    - [queryRound(feed)](#queryRound)
+    - [submitNewJobIndex(index, requestType)](#submitNewJobIndex)
+    - [makeController()](#makeController)
+    - [getOraclesInvitations()](#getOraclesInvitationsMiddleware)
+    - [pushPrice(price, feed, round, from)](#pushPrice)
+    - [startBridge(port)](#startBridge)
+      - [POST /adapter](#postadapter)
+      - [POST /jobs](#postjobs)
+      - [DELETE /jobs/:id](#deljobs)
+    - [middleware()](#middlewarefunc)
+  - [monitor.js](#monitorjs)
+    - [Environment Variables](#envvarsmonitor)
+    - [Metrics](#metrics)
+    - [readOracleAddresses()](#readOracleAddresses)
+    - [readMonitoringState()](#readMonitoringState)
+    - [getOraclesInvitations()](#getOraclesInvitations)
+    - [updateMetrics(oracleName, oracle, feed, value, id, actualPrice, lastRound)](#updateMetrics)
+    - [updateBalanceMetrics(oracleName, oracle, feed, value)](#updateBalanceMetrics)
+    - [queryPrice(feed)](#queryPriceMonitor)
+    - [getOffersAndBalances(follower, oracle)](#getOffersAndBalances)
+    - [getLatestPrices(oracle, oracleDetails, state)](#getLatestPrices)
+    - [startServer()](#startServer)
+    - [monitor()](#monitorfunc)
+- [Considerations](#considerations)
+  - [Why not use a Chainlink cron job?](#consid1)
+  - [Why does the middleware consist of an exposed API and two intervalled functions?](#consid2)
+  - [Why does the middleware only consider the last few offers when making checks?](#consid3)
+  - [Intervals](#consid4)
+## Oracle Network Structure
+
+<img src="images/struct.png"></img>
+<br>
+
+Each node operator has to run:
+- A Chainlink node hosted locally in Docker that queries Chainlink adapters
+- Chainlink adapters hosted beside the Chainlink node that act as middleware between the node as various 3rd party APIs
+- Middleware which initiates jobs on the Chainlink node through its API if a round hasn't been created in 1 minute on the on chain aggregator contract. Rounds are also created if the oracle operator's current price is deviated past X% from the latest on-chain median price. Oracle operators are not allowed to create 2 rounds subsequently to prevent spamming.
+- An Agoric node hosted on their own infrastructure to allow the middleware / oracle proxy to communicate and broadcast transactions reliably to the network.
+
+
+## Smart Contract Details
+
+Smart Contract - https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/src/price/priceAggregatorChainlink.js
```

Comment:
not found, use a commit hash when posting a github link

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/d85360c17cb9f44394ff03ce5df97da7c44ef58a